### PR TITLE
DEVPROD-7739: auto-quarantine and alert for static hosts

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-7739

### Description
* Automatically quarantine a static host if it system-fails three tasks in a row. (This behavior already used to exist but EVG-2742 mysteriously removed it for unknown reasons. I think it's fine to reintroduce.)
* Send a Splunk log when a static host is quarantined. I refactored the job so that it's less weird and more in line with how we typically write jobs (i.e. not persisting an entire copy of the host doc in the job). When this logs, [an alert will ping `#devprod-runtime-environment-alerts`](https://mongodb.splunkcloud.com/en-US/app/search/alert?s=%2FservicesNS%2Fnobody%2Fsearch%2Fsaved%2Fsearches%2FEvergreen%253A%2520Static%2520Host%2520Quarantined) that a static host is quarantined.
* Clean up some old next/end task tests so they're more in line with current coding style standards. While fixing the tests, I also drive-by fixed a tiny issue where the end task route returns the wrong type of payload in one case.

### Testing
* Added unit tests.
* Tested in staging that quarantining a host created the Splunk log.
* After the deploy, will verify that static host quarantine alerts go through.

### Documentation
N/A

### CODE REVIEW

File: rest/route/host_agent.go [hunk 1]
```diff
@@ -1238,22 +1238,14 @@ func (h *hostAgentEndTask) Run(ctx context.Context) gimlet.Responder {
 		endTaskResp.ShouldExit = true
 	}
 
-	// we should disable hosts and prevent them from performing
-	// more work if they appear to be in a bad state
-	// (e.g. encountered 5 consecutive system failures)
+	// Disable hosts and prevent them from performing more work if they have
+	// system failed many tasks in a row.
```
Comment (hadjri) [inline rest/route/host_agent.go:1242]: nit: many -> consecutiveSystemFailureThreshold

File: rest/route/host_agent.go [hunk 2]
```diff
@@ -1238,22 +1238,14 @@ func (h *hostAgentEndTask) Run(ctx context.Context) gimlet.Responder {
...
 	if event.AllRecentHostEventsMatchStatus(ctx, currentHost.Id, consecutiveSystemFailureThreshold, evergreen.TaskSystemFailed) {
 		msg := "host encountered consecutive system failures"
```
Comment (hadjri) [inline rest/route/host_agent.go:1244]: nit: can we format this string with `consecutiveSystemFailureThreshold`

**Review by hadjri — APPROVED** [review]

File: rest/route/host_agent.go (hunk 1 above)
Comment (Kimchelly) [reply rest/route/host_agent.go:1242]: I think the comment is fine as-is, you can see the `consecutiveSystemFailureThreshold` immediately below and the comment is only there to state the general reasoning of why this special behavior is here, not restate what the code is doing.

File: rest/route/host_agent.go (hunk 2 above)
Comment (Kimchelly) [reply rest/route/host_agent.go:1244]: Done.

**Pull request merged**